helper/reader.go

```diff
@@ -0,0 +1,51 @@
+package helper
+
+import (
+	"io"
+)
+
+type MultiDimByteReader struct {
```

Comment:
I think, we need to add tests for reader and utils

Reply:
Done with love.